### ISSUE TITLE
unable to reproduce example

### ISSUE THREAD

**Issue by adpala** [KlugerLab/FIt-SNE#59]
While trying to reproduce your example code in atom, the result of the first figure looks nothing like yours. I don't know where I might be changing things. I'm doing this in Windows. Here is how my code looks like:

import numpy as np
import sys
import os
import pylab as plt
sys.path.append('path to /FIt-SNE')
from fast_tsne import fast_tsne

from keras.datasets import mnist
(x_train, y_train), (x_test, y_test) = mnist.load_data()
x_train = x_train.reshape(60000, 784).astype('float64') / 255
x_test = x_test.reshape(10000, 784).astype('float64') / 255
X = np.concatenate((x_train, x_test))
y = np.concatenate((y_train, y_test))
print(X.shape)
X = X - X.mean(axis=0)
U, s, V = np.linalg.svd(X, full_matrices=False)
X50 = np.dot(U, np.diag(s))[:, :50]

print(X50.shape)

PCAinit = X50[:, :2] / np.std(X50[:, 0]) * 0.0001
col = np.array(['#a6cee3', '#1f78b4', '#b2df8a', '#33a02c', '#fb9a99',
                '#e31a1c', '#fdbf6f', '#ff7f00', '#cab2d6', '#6a3d9a'])

Z = fast_tsne(X50, perplexity=50, seed=42)

print(Z.shape)

plt.figure(figsize=(5, 5))
plt.scatter(Z[:, 0], Z[:, 1], c=col[y], s=1)
plt.tight_layout()
plt.show()

![figure_1](https://user-images.githubusercontent.com/43853502/49082914-5c4fc200-f24b-11e8-8b34-d3e376fc771f.png)

**Comment by dkobak** [comment]
This is weird. Can you plot the first two dimensions of X50 with the same code?

**Comment by dkobak** [comment]
Why don't you use `plt.scatter(X50[:, 0], X50[:, 1], c=col[y], s=1)`? This is hard to interpret.

Nevertheless, it seems that PCA representation is fine. I have no idea what's going on with the `fast_tsne` call. Can you post the full console output that you get from the binary?

**Comment by adpala** [comment]
It seems that Atom did not show me that running FItSNE.exe was failing. It shows me the following error:

![image](https://user-images.githubusercontent.com/43853502/49093862-4e0e9f80-f265-11e8-850a-acb256b3c94f.png)

I'm also having trouble trying to install the pyFIt-SNE from https://github.com/KlugerLab/pyFIt-SNE, this two problems might be related (?) here is the link to that other issue: https://github.com/KlugerLab/pyFIt-SNE/issues/12

Here is the colored version of the scatter plot anyway:

![figure_1](https://user-images.githubusercontent.com/43853502/49093956-82825b80-f265-11e8-8208-8a6fd2ae3a19.png)

**Comment by dkobak** [comment]
The .dll is right in here https://github.com/KlugerLab/FIt-SNE/releases/download/v1.0.0/FItSNE-Windows-1.0.0.zip (assuming that's where you took the .exe file).

**Comment by adpala** [comment]
After reading again the installation description, I noticed you mention some dlls files that have to be copied into the bin when doing the compilation on their own, which I thought I did not had to do, because the previous instructions only mentioned copying the .exe file in the bin folder.
The function now runs and gives me the following terminal output, but because it is giving a 1000 x 2 matrix, I had to disable the colors option in the plot, due to size differences:


(leap) Z:\#Common\adrian\Workspace\python_scripts\sandbox>python testing_tsne.py
Z:\#Common\adrian\Workspace\python_scripts\sandbox
Using TensorFlow backend.
(70000, 784)
=============== t-SNE v1.0.0 ===============
fast_tsne data_path: data.dat
fast_tsne result_path: result.dat
fast_tsne nthreads: 8
Read the following parameters:
         n 1000 by d 50 dataset, theta 0.500000,
         perplexity 30.000000, no_dims 2, max_iter 1000,
         stop_lying_iter 250, mom_switch_iter 250,
         momentum 0.500000, final_momentum 0.800000,
         learning_rate 200.000000, K -1, sigma -1.000000, nbody_algo 2,
         knn_algo 1, early_exag_coeff 12.000000,
         no_momentum_during_exag 0, n_trees 50, search_k 4500,
         start_late_exag_iter -1, late_exag_coeff -1.000000
         nterms 3, interval_per_integer 1.000000, min_num_intervals 50
Read the 1000 x 50 data matrix successfully. X[0,0] = -0.663666
Using current time as random seed...
Will use momentum during exaggeration phase
Computing input similarities...
Using perplexity, so normalizing input data (to prevent numerical problems)
Using perplexity, not the manually set kernel width.  K (number of nearest neighbors) and sigma (bandwidth) parameters are going to be ignored.
Using ANNOY for knn search, with parameters: n_trees 50 and search_k 4500
Going to allocate memory. N: 1000, K: 90, N*K = 90000
Building Annoy tree...
Done building tree. Beginning nearest neighbor search...
parallel (8 threads):
[============================================================] 100% 0.097s
Symmetrizing...
Randomly initializing the solution.
Exaggerating Ps by 12.000000
Input similarities computed (sparsity = 0.136632)!
Learning embedding...
Using FIt-SNE approximation.
Iteration 50 (50 iterations in 0.50 seconds), cost 3.194069
Iteration 100 (50 iterations in 0.45 seconds), cost 3.217557
Iteration 150 (50 iterations in 0.45 seconds), cost 3.284607
Iteration 200 (50 iterations in 0.45 seconds), cost 3.215625
Iteration 250 (50 iterations in 0.44 seconds), cost 3.207101
Unexaggerating Ps by 12.000000
Iteration 300 (50 iterations in 0.45 seconds), cost 2.855076
Iteration 350 (50 iterations in 0.44 seconds), cost 2.682906
Iteration 400 (50 iterations in 0.43 seconds), cost 2.663008
Iteration 450 (50 iterations in 0.43 seconds), cost 2.631070
Iteration 500 (50 iterations in 0.43 seconds), cost 2.603290
Iteration 550 (50 iterations in 0.42 seconds), cost 2.616587
Iteration 600 (50 iterations in 0.44 seconds), cost 2.598216
Iteration 650 (50 iterations in 0.44 seconds), cost 2.600974
Iteration 700 (50 iterations in 0.47 seconds), cost 2.602294
Iteration 750 (50 iterations in 0.44 seconds), cost 2.608419
Iteration 800 (50 iterations in 0.49 seconds), cost 2.606481
Iteration 850 (50 iterations in 0.51 seconds), cost 2.588602
Iteration 900 (50 iterations in 0.63 seconds), cost 2.591058
Iteration 950 (50 iterations in 0.65 seconds), cost 2.591046
Iteration 1000 (50 iterations in 0.64 seconds), cost 2.562554
Wrote the 1000 x 2 data matrix successfully.
Done.

**Comment by dkobak** [comment]
OK, this might actually be a bug, but I'm not quite sure where.

```
=============== t-SNE v1.0.0 ===============
fast_tsne data_path: data.dat
fast_tsne result_path: result.dat
fast_tsne nthreads: 8
Read the following parameters:
n 1000 by d 50 dataset, theta 0.500000,
```

It says right here that the data it read in had n=1000, but in Python your data matrix has n=70000.

**Comment by dkobak** [comment]
Are you using this wrapper? https://github.com/KlugerLab/FIt-SNE/blob/master/fast_tsne.py

If so, can you add `print(n)` after line 62? https://github.com/KlugerLab/FIt-SNE/blob/master/fast_tsne.py#L62

**Comment by adpala** [comment]
it prints 70000

**Comment by adpala** [comment]
could it be somehow related to max_iter = 1000 by default?

**Comment by dkobak** [comment]
I don't know. Really weird. I have to go now, will come back to this later.

**Comment by adpala** [comment]
Just to add more information, even if I give n < 1000 or d < 50, it still reads:  n 1000 by d 50 dataset, theta 0.500000.

**Comment by dkobak** [comment]
Wait a sec, you had  `X = np.random.randn(1000, 50)` in the initial post.

**Comment by adpala** [comment]
Nevermind, I found the problem. I changed the paths to the result.dat files when trying to solve the original problem, and forgot to put it back in order. Now it is running fine

**Comment by adpala** [comment]
sorry for the inconvenience
